lib/rubanok/bad_value_error.rb

```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+module Rubanok
+  class BadValueError < StandardError; end
```

Comment:
First, "bad" is a bad word and doesn't tell anything meaningful. Let's call it `UnexpectedValueError`.

Second, why putting it in a separate file? It's a `Matching`-specific error, let's define it there (but under the top-level namespace, i.e. `Rubanok::UnexpectedValueError`).